File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int>mp;
        for(int num : arr){
            mp[num]++;
        }
        unordered_set<int>set;

        for(auto it : mp){
            if(set.count(it.second)){
                return false;
            }
            set.insert(it.second);
        }
        return true;
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        int n=arr.length;
        HashMap<Integer,Integer>mp= new HashMap<>();
        HashSet<Integer>st = new HashSet<>();
 
        for(int i=0;i<n;i++){
            mp.put(arr[i],mp.getOrDefault(arr[i],0)+1);
        }
        for(var ch : mp.entrySet()){
            st.add(ch.getValue());
        }
        return mp.size()==st.size();
    }
}